src/custom/pages/Profile/index.tsx

```diff
@@ -182,13 +189,24 @@ export default function Profile() {
     return content
   }, [isSwapConfirmed, isSwapPending])
 
+  // Fixes the issue with change in status after swap confirmation
+  // Makes sure to wait 2 blocks after confirmation to enable the swap button again
   useEffect(() => {
-    if (isSwapConfirmed && hasVestedBalance) {
-      setTimeout(() => {
-        setSwapVCowStatus(SwapVCowStatus.INITIAL)
-      }, 5000)
+    if (isSwapConfirmed && !confirmationBlock) {
+      setConfirmationBlock(blockNumber)
+      setShouldUpdate(true)
+    }
+
+    if (!confirmationBlock || !blockNumber) {
+      return
+    }
+
+    if (isSwapConfirmed && blockNumber - confirmationBlock > 2 && !vested?.equalTo(0)) {
```

Comment:
won't the `!` `vested?` here also break like the same code above where u do `vested && !vested?.equalTo(0)`?

Reply:
Good point, updated :+1: